geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -1528,7 +1541,7 @@ public void updatePRConfig(PartitionRegionConfig prConfig, boolean putOnlyIfUpda
             this.prRoot.get(colocatedRegion.getRegionIdentifier());
         if (parentConf.isColocationComplete() && parentConf.hasSameDataStoreMembers(prConfig)) {
           colocationComplete = true;
-          prConfig.setColocationComplete();
+          prConfig.setColocationComplete(this);
```

Comment:
Would it be better to call executeColocationCallbacks here? That way you do not need to change PartitionRegionConfig and do not need to pass "this" to it.